shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/token/generator/EncryptAlterTableTokenGenerator.java

```diff
@@ -135,6 +150,27 @@ public boolean isGenerateSQLToken(final SQLStatementContext sqlStatementContext)
         return result;
     }
     
+    private Collection<SQLToken> getAddColumnTokensWithConfigDataType(final String tableName, final String columnName, final AddColumnDefinitionSegment addColumnDefinitionSegment, 
+                                                                      final ColumnDefinitionSegment columnDefinitionSegment) {
+        Collection<SQLToken> result = new LinkedList<>();
+        EncryptColumn encryptColumn = getEncryptColumn(tableName, columnName);
+        result.add(new RemoveToken(columnDefinitionSegment.getStartIndex(), columnDefinitionSegment.getStopIndex()));
+        result.add(new EncryptConfigDataTypeToken(columnDefinitionSegment.getStopIndex() + 1, columnDefinitionSegment.getStopIndex(),
+                encryptRule.getCipherColumn(tableName, columnName), encryptColumn.getCipherDataType().getTypeName()));
+        Optional<String> assistedQueryColumn = encryptRule.findAssistedQueryColumn(tableName, columnName);
+        assistedQueryColumn.map(assisted -> new EncryptConfigDataTypeToken(addColumnDefinitionSegment.getStopIndex() + 1, 
+                addColumnDefinitionSegment.getStopIndex(), ", ADD COLUMN " + assisted, encryptColumn.getAssistedQueryDataType().getTypeName())).ifPresent(result::add);
+        Optional<String> plainColumn = encryptRule.findPlainColumn(tableName, columnName);
+        plainColumn.map(plain -> new EncryptConfigDataTypeToken(addColumnDefinitionSegment.getStopIndex() + 1, 
+                addColumnDefinitionSegment.getStopIndex(), ", ADD COLUMN " + plain, encryptColumn.getPlainDataType().getTypeName())).ifPresent(result::add);
+        return result;
+    }
+    
+    private EncryptColumn getEncryptColumn(final String tableName, final String columnName) {
```

Comment:
@tuichenchuxin Can we move this method to EncryptRule?

Reply:
> @tuichenchuxin Can we move this method to EncryptRule?

Ok, Done.